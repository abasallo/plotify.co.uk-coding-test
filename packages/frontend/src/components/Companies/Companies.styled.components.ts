import styled from 'styled-components'

import { Card, CardActions, Grid } from '@material-ui/core'

import MaterialUIContainer from '@material-ui/core/Container'

export const Container = styled(MaterialUIContainer)`
  display: flex;
  flex-direction: column;
`

export const CompaniesGrid = styled(Grid)`
  flex-grow: 1;
`
export const CompanyComponent = styled(Card)`
  cursor: pointer;
  padding: 10px;
  margin: 10px;
  min-height: 100px;
  text-align: center;
`

export const CompanyActions = styled(CardActions)`
  text-align: right;
`
