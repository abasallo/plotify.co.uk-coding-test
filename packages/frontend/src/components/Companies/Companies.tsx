import React, { useEffect, useState } from 'react'
import { useNavigate } from 'react-router-dom'

import { Grid } from '@material-ui/core'

import { CompanyComponent, CompanyActions, CompaniesGrid, Container } from './Companies.styled.components'

import { Company } from 'plotify.co.uk-coding-test-shared'

import { getCompanies } from '../../services/company'

import constants from '../../modules/constants'

const Companies = () => {
  const navigate = useNavigate()

  const [companies, setCompanies] = useState<Company[]>([])

  useEffect(() => {
    const dataFetch = async () => setCompanies(await getCompanies())
    dataFetch()
  }, [])

  const handleDetailsClick = (companyId: number) => navigate(`${constants.ROUTE_URL_COMPANY}/${companyId}`)

  return (
    <Container>
      <CompaniesGrid container spacing={2}>
        {companies.map((company: Company) => (
          <Grid item xs={12} sm={6} md={4} lg={3} key={company.id}>
            <CompanyComponent onClick={() => handleDetailsClick(company.id)}>
              <CompanyActions disableSpacing>{company.name}</CompanyActions>
            </CompanyComponent>
          </Grid>
        ))}
      </CompaniesGrid>
    </Container>
  )
}

export default Companies
