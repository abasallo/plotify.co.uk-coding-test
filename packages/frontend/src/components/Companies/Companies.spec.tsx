import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

import Companies from './Companies'

describe('Companies Component', () => {
  test('Renders properly', async () => expect(renderer.create(
    <BrowserRouter>
      <Companies />
    </BrowserRouter>
  ).toJSON()).toMatchSnapshot())
})
