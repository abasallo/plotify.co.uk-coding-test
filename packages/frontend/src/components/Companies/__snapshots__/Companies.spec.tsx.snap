// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Companies Component Renders properly 1`] = `
<div
  className="MuiContainer-root sc-beqWaB emoJJV MuiContainer-maxWidthLg"
>
  <div
    className="MuiGrid-root sc-gueYoa gcjcJu MuiGrid-container MuiGrid-spacing-xs-2"
  />
</div>
`;
